examples/notebooks/beam-ml/automatic_model_refresh.ipynb

```diff
@@ -1,605 +1,652 @@
+        "\n",
+        "# The Dataflow temp location. This location is used to store temporary files or intermediate results before outputting to the sink.\n",
+        "options.view_as(GoogleCloudOptions).temp_location = '%s/temp' % dataflow_gcs_location\n",
+        "\n",
+        "options.view_as(SetupOptions).save_main_session = True\n",
+        "\n",
+        "# Launching Dataflow with only one worker might result in processing delays due to\n",
+        "# initial input processing. This could further postpone the side input model updates.\n",
+        "# To expedite the model update process, it's recommended to set num_workers>1.\n",
+        "# https://github.com/apache/beam/issues/28776\n",
+        "options.view_as(WorkerOptions).num_workers = 5"
+      ],
+      "metadata": {
+        "id": "wWjbnq6X-4uE",
+        "outputId": "2125c017-dfd4-4402-f02a-8469f67409a8"
+      },
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Install the `tensorflow` and `tensorflow_hub` dependencies on Dataflow. Use the `requirements_file` pipeline option to pass these dependencies."
+      ],
+      "metadata": {
+        "id": "HTJV8pO2Wcw4"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# In a requirements file, define the dependencies required for the pipeline.\n",
+        "deps_required_for_pipeline = ['tensorflow>=2.12.0', 'tensorflow-hub>=0.10.0', 'Pillow>=9.0.0']\n",
+        "requirements_file_path = './requirements.txt'\n",
+        "# Write the dependencies to the requirements file.\n",
+        "with open(requirements_file_path, 'w') as f:\n",
+        "  for dep in deps_required_for_pipeline:\n",
+        "    f.write(dep + '\\n')\n",
+        "\n",
+        "# Install the pipeline dependencies on Dataflow.\n",
+        "options.view_as(SetupOptions).requirements_file = requirements_file_path"
+      ],
+      "metadata": {
+        "id": "lEy4PkluWbdm",
+        "outputId": "a594833e-4ffd-4abc-a0bc-90e755ee4f8a"
+      },
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Use the TensorFlow model handler\n",
+        " This example uses `TFModelHandlerTensor` as the model handler and the `resnet_101` model trained on [ImageNet](https://www.image-net.org/).\n",
+        "\n",
+        " Download the model from [Google Cloud Storage](https://storage.googleapis.com/tensorflow/keras-applications/resnet/resnet101_weights_tf_dim_ordering_tf_kernels.h5) (link downloads the model), and place it in the directory that you want to use to update your model.\n",
```

Comment:
I found that just downloading the model naively didn't work and I got an error about it being saved in an improper format. This can be demonstrated by using the same command our model handler uses for loading: `tf.keras.models.load_model('path/to/resnet101_weights_tf_dim_ordering_tf_kernels.h5')`

The fix I found was to load the model and then save it with keras. So:

```
model = tf.keras.applications.resnet.ResNet101()
model.save('path/to/resnet101_weights_tf_dim_ordering_tf_kernels.keras')
model = tf.keras.applications.resnet.ResNet152()
model.save('path/to/resnet152_weights_tf_dim_ordering_tf_kernels.keras')
```

(and then using the `.keras` extension elsewhere). I think the `.h5` files just contain the model weights